src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```
@@ -0,0 +1,755 @@
+{
+  std::cout << "----------------------------------------------------\n";
+  pretty_print_platform_info(pt_device, vbnv);
+  std::cout << "----------------------------------------------------\n";
+
+  std::stringstream action_list;
+
+  if (!pt_device.get<bool>("platform.status.shell"))
+    action_list << boost::format("  [%s] : Program base (FLASH) image\n") % pt_device.get<std::string>("platform.bdf");
+
+  if (!pt_device.get<bool>("platform.status.sc") && !pt_device.get<bool>("platform.status.is_factory") && !pt_device.get<bool>("platform.status.is_recovery"))
+    action_list << boost::format("  [%s] : Program Satellite Controller (SC) image\n") % pt_device.get<std::string>("platform.bdf");
+
+  if (!action_list.str().empty()) {
+    std::cout << "Actions to perform:\n" << action_list.str();
+    std::cout << "----------------------------------------------------\n";
+  }
+}
+
+static bool
+are_shells_equal(const DSAInfo& candidate, const DSAInfo& current)
+{
+  if (current.dsaname().empty())
+    throw std::runtime_error("Current shell name is empty.");
+
+  return ((candidate.dsaname() == current.dsaname()) && candidate.matchId(current));
+}
+
+static bool
+are_scs_equal(const DSAInfo& candidate, const DSAInfo& current)
+{
+  if (current.dsaname().empty())
+    throw std::runtime_error("Current shell name is empty.");
+
+  return ((current.bmcVer.compare("INACTIVE") == 0) || (candidate.bmc_ver() == current.bmc_ver()));
+}
+
+static bool
+update_sc(unsigned int boardIdx, DSAInfo& candidate)
+{
+  Flasher flasher(boardIdx);
+
+  // Determine if the SC images are the same
+  bool same_bmc = false;
+  DSAInfo current = flasher.getOnBoardDSA();
+  if (!current.dsaname().empty())
+    same_bmc = are_scs_equal(candidate, current);
+
+  // -- Some DRCs (Design Rule Checks) --
+  // Is the SC present
+  if (current.bmc_ver().empty() || candidate.bmc_ver().empty()) {
+    std::cout << "INFO: Satellite controller is not present.\n";
+    return false;
```

Exceptions should be thrown, instead of using return values.